code/obj/item/misc_weapons.dm

```diff
@@ -2207,3 +2211,34 @@ obj/item/whetstone
 	New()
 		..()
 		src.setItemSpecial(/datum/item_special/rangestab)
+
+
+
+
+/obj/item/gang_machete
+	name = "machete"
+	desc = "A hefty, unbalanced blade. Wielding it fills you with thoughts of savagery."
+	icon = 'icons/obj/items/weapons.dmi'
+	icon_state = "machete"
+	item_state = "welder_machete"
+	inhand_image_icon = 'icons/mob/inhand/hand_weapons.dmi'
+	hit_type = DAMAGE_CUT
+	flags = FPRINT | USEDELAY
+	force = 25
+	click_delay = 16 DECI SECONDS //unbalanced blade
+	throwforce = 5
+	throw_speed = 1
+	throw_range = 5
+	contraband = 4
+	w_class = W_CLASS_NORMAL
+	tool_flags = TOOL_CUTTING
+	attack_verbs = "slashes"
+	hitsound = 'sound/impact_sounds/Blade_Small_Bloody.ogg'
+
+	New()
+		..()
+		src.AddComponent(/datum/component/bloodflick)
+		src.setItemSpecial(/datum/item_special/massacre)
+	on_spin_emote(mob/user)
+		SEND_SIGNAL(src, COMSIG_ITEM_TWIRLED, user, src)
+		animate_spin(src, prob(50) ? "L" : "R", 1, 0)
```

Comment:
What is this actually doing? It just seems to be recreating some of the base behaviour of the `twirl` emote (and sending the signal twice). If you want the behaviour of the twirl emote then just make the user do the emote instead.